server/src/test/java/io/crate/integrationtests/ForeignDataWrapperITest.java

```diff
@@ -316,4 +316,48 @@ public void test_can_use_joins_on_foreign_tables() throws Exception {
             "Dom| CH"
         );
     }
+
+    @Test
+    public void test_mask_foreign_password_from_user_mapping_options_table() {
+        execute("create user user1 with (password = 'user1pw')");
+        execute("create user user2 with (password = 'user2pw')");
+        execute("grant al to user1");
+        execute("grant al to user2");
+
+        var roles = cluster().getInstance(Roles.class);
+        Role user1 = roles.findUser("user1");
+        sqlExecutor.executeAs("""
+            CREATE SERVER my_postgresql
+            FOREIGN DATA WRAPPER jdbc
+            OPTIONS (url 'jdbc:postgresql://example.com:5432/');
+            """, user1);
+        sqlExecutor.executeAs("""
+            CREATE FOREIGN TABLE doc.remote_documents (name text)
+            SERVER my_postgresql
+            OPTIONS (schema_name 'public', table_name 'documents');
+            """, user1);
+        sqlExecutor.executeAs("""
+            CREATE USER MAPPING FOR user1
+            SERVER my_postgresql
+            OPTIONS ("user" 'foreign-user', password 'foreign-pw');
+            """, user1);
+
+        // user1 can see the pw because user1 is being mapped
+        var response = sqlExecutor.executeAs("select * from information_schema.user_mapping_options where option_name = 'password'", user1);
+        assertThat(response).hasRows("user1| crate| my_postgresql| password| foreign-pw");
+
+        // user2 cannot see the pw because user2 is not being mapped nor is a superuser
+        response = sqlExecutor.executeAs("select * from information_schema.user_mapping_options where option_name = 'password'",
+            roles.findUser("user2"));
+        assertThat(response).hasRows("user1| crate| my_postgresql| password| NULL");
+
+        // superuser can see the pw
+        response = execute("select * from information_schema.user_mapping_options where option_name = 'password'");
+        assertThat(response).hasRows("user1| crate| my_postgresql| password| foreign-pw");
+
+        execute("drop user mapping for user1 server my_postgresql");
+        assertThat(execute("select * from information_schema.user_mapping_options where option_name = 'password'")).isEmpty();
+        execute("drop foreign table doc.remote_documents");
+        execute("drop server my_postgresql");
```

Comment:
```suggestion
```

Could call the server `pg` and the users `trillian` and `arthur` to have the teardown method do this:

https://github.com/crate/crate/blob/0f47b8185b6681f06f3798c5392d11646a377d46/server/src/test/java/io/crate/integrationtests/ForeignDataWrapperITest.java#L41-L46

Would have the advantage that a test failure here doesn't cause the other tests to fail as well.